python/ray/tune/examples/mnist_pytorch_lightning.py

```diff
@@ -0,0 +1,252 @@
+
+
+# __tune_train_checkpoint_begin__
+def train_mnist_tune_checkpoint(config, checkpoint=None):
+    trainer = pl.Trainer(
+        max_epochs=10,
+        progress_bar_refresh_rate=0,
+        callbacks=[CheckpointCallback(),
+                   TuneReportCallback()])
+    if checkpoint:
+        # Currently, this leads to errors:
+        # model = LightningMNISTClassifier.load_from_checkpoint(
+        #     os.path.join(checkpoint, "checkpoint"))
+        # Workaround:
+        ckpt = pl_load(
+            os.path.join(checkpoint, "checkpoint"),
+            map_location=lambda storage, loc: storage)
+        model = LightningMNISTClassifier._load_model_state(ckpt, config=config)
+        trainer.current_epoch = ckpt["epoch"]
+    else:
+        model = LightningMNISTClassifier(
+            config=config, data_dir=config["data_dir"])
+
+    trainer.fit(model)
+# __tune_train_checkpoint_end__
+
+
+# __tune_asha_begin__
+def tune_mnist_asha():
+    data_dir = mkdtemp(prefix="mnist_data_")
+    LightningMNISTClassifier.download_data(data_dir)
+    config = {
+        "layer_1_size": tune.choice([32, 64, 128]),
+        "layer_2_size": tune.choice([64, 128, 256]),
+        "lr": tune.loguniform(1e-4, 1e-1),
+        "batch_size": tune.choice([32, 64, 128]),
+        "data_dir": data_dir
+    }
+    scheduler = ASHAScheduler(
+        metric="loss",
+        mode="min",
+        max_t=10,
+        grace_period=1,
+        reduction_factor=2)
+    reporter = CLIReporter(
+        parameter_columns=["layer_1_size", "layer_2_size", "lr", "batch_size"],
+        metric_columns=["loss", "mean_accuracy", "training_iteration"])
+    tune.run(
+        train_mnist_tune,
+        resources_per_trial={"cpu": 1},
+        config=config,
+        num_samples=10,
+        scheduler=scheduler,
+        progress_reporter=reporter)
+    shutil.rmtree(data_dir)
+# __tune_asha_end__
+
+
+# __tune_pbt_begin__
+def tune_mnist_pbt():
+    data_dir = mkdtemp(prefix="mnist_data_")
+    LightningMNISTClassifier.download_data(data_dir)
+    config = {
+        "layer_1_size": tune.choice([32, 64, 128]),
+        "layer_2_size": tune.choice([64, 128, 256]),
+        "lr": 1e-3,
+        "batch_size": 64,
+        "data_dir": data_dir
+    }
+    scheduler = PopulationBasedTraining(
+        time_attr="training_iteration",
+        metric="loss",
+        mode="min",
+        perturbation_interval=4,
+        hyperparam_mutations={
+            "lr": lambda: tune.loguniform(1e-4, 1e-1).func(None),
+            "batch_size": [32, 64, 128]
+        })
+    reporter = CLIReporter(
+        parameter_columns=["layer_1_size", "layer_2_size", "lr", "batch_size"],
+        metric_columns=["loss", "mean_accuracy", "training_iteration"])
+    tune.run(
+        train_mnist_tune_checkpoint,
+        resources_per_trial={"cpu": 1},
+        config=config,
+        num_samples=10,
+        scheduler=scheduler,
+        progress_reporter=reporter)
+    shutil.rmtree(data_dir)
```

Comment:
any reason we need to rmtree here?

Reply:
We download the MNIST data to a (currently not reused) temporary directory, so I'd like to free the space afterwards